File: InterviewPratice/Java/FirstDuplicate.java
import java.util.Set;
import java.util.HashSet;

final class FirstDuplicate {

	int firstDuplicate(int[] a) {
	    Set<Integer> set = new HashSet<>();
	    
	    for(int i : a) if(!set.add(i)) return i;
	    
	    return -1;
	}
}

File: InterviewPratice/Javascript/firstDuplicate.js
const firstDuplicate = (a) => {
    for(let i = 0; i < a.length; i++) {
        if(a[Math.abs(a[i]) - 1] < 0) return Math.abs(a[i]);
        a[Math.abs(a[i]) - 1] = -a[Math.abs(a[i]) - 1];
    }
    return -1;
}